### PR TITLE
Arm backend batchnorm2d fixes and tests

### PR DESCRIPTION
**Pull request by zingo**
Batchnorm2d did not handle weight and bias

### CODE REVIEW

**Review by digantdesai — APPROVED** [review]

File: backends/arm/test/ops/test_batch_norm.py [hunk 1]
```diff
@@ -0,0 +1,705 @@
...
+                self.batch_norm_2d.bias = torch.nn.Parameter(bias)
+            if running_mean is not None:
+                self.batch_norm_2d.running_mean = running_mean
+            if running_var is not None:
+                self.batch_norm_2d.running_var = running_var
+
+        def forward(self, x):
+            return self.batch_norm_2d(x)
+
+    def _test_batchnorm2d_tosa_MI_pipeline(
```
[the remark this answers is not in the document]
Comment (zingo) [reply backends/arm/test/ops/test_batch_norm.py:527]: maybe, depends on how you mean, we have now followed the flow in out unit tests where we have 3 different methods
_test_*_tosa_MI_pipeline()
_test_*_tosa_BI_pipeline()
_test_*_u55_BI_pipeline()

Where tosa onle test tosa layers and MI and BI are tosa profiles (MI - float and BI - Integers) and u55 also involves creating output all the way to u55 invoking the next step (the vela compiler) all tree of this invokes some different steps and profiles and we right now at least thing it makes sense to keep them differently

As for the  _test_batchnorm2d_tosa_MI_pipeline() and _test_batchnorm2d_no_stats_tosa_MI_pipeline() split added in this unit test file I just decided to split it as it ends up as 2 different aten's but Im also unsure if we even expect to see torch.ops.aten._native_batch_norm_legit_no_stats at all here? e.g. is thus used only for training? If so maybe we could remove those test or handle them differently. I just added it to try to test all combinations of input flags :)

File: backends/arm/test/ops/test_batch_norm.py [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+            ArmTester(
+                module,
+                inputs=test_data,
+                compile_spec=common.get_tosa_compile_spec(),
+            )
+            .export()
+            .check_count(
+                {"torch.ops.aten._native_batch_norm_legit_no_training.default": 1}
+            )
+            .check_not(["torch.ops.quantized_decomposed"])
```
[the remark this answers is not in the document]
Comment (zingo) [reply backends/arm/test/ops/test_batch_norm.py:540]: Thanks for the review!
To my understanding  this should complain if there is none e.g. if count is not "1" 
```
.check_count(
                {"torch.ops.aten._native_batch_norm_legit_no_training.default": 1} 
```

Maybe I misunderstand it or is it something else you are thinking about? If so I'm happy to change the code.

**Comment by zingo** [comment]
I see this is "imported" anything more you want me to do or should I just wait :)

File: backends/arm/test/ops/test_batch_norm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (digantdesai) [reply backends/arm/test/ops/test_batch_norm.py:527]: I meant only for code-deduplication. We may still keep the wrapper functions to make it more readable.

File: backends/arm/test/ops/test_batch_norm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (digantdesai) [reply backends/arm/test/ops/test_batch_norm.py:540]: Something like [this](https://github.com/pytorch/executorch/blob/3df8695a8816b9002a9b2dade33ef73629b919c1/backends/xnnpack/test/ops/pow.py#L34) where we know the op is consumed by the delegate.

**Comment by zingo** [comment]
Sorry didn't have time to look into this patch today.

**Comment by zingo** [comment]
Some comments fixed in https://github.com/pytorch/executorch/pull/3658
